tests/test_graphql_api/test_application_event_schedule/test_approve.py

```diff
@@ -184,3 +205,331 @@ def test_cannot_approve_if_application_is_not_in_unallocated(graphql):
+    # given:
+    # - There is an allocatable application event schedule
+    # - A superuser is using the system
+    application = ApplicationFactory.create_application_ready_for_allocation()
+    graphql.login_user_based_on_type(UserType.SUPERUSER)
+
+    # when:
+    # - The user tries to allocate the schedule, but for a reservation unit
+    #   not in the event's event reservation units.
+    input_data = approve_data(application, begin="10:00:00", end="12:00:00")
+    input_data["allocatedReservationUnit"] = ReservationUnitFactory.create().pk
+    response = graphql(APPROVE_MUTATION, input_data=input_data)
+
+    # then:
+    # - The response complains about the reservation unit not being included.
+    assert response.field_error_messages("allocatedReservationUnit") == [
+        "Cannot allocate schedule for this reservation unit. Reservation unit is not included in the event.",
+    ]
+
+
+def test_approve_schedule__cannot_approve_two_schedules_for_same_day(graphql):
+    # given:
+    # - There is an allocatable application event schedule
+    # - A superuser is using the system
+    application = ApplicationFactory.create_application_ready_for_allocation(events_per_week=3, pre_allocated=True)
+    graphql.login_user_based_on_type(UserType.SUPERUSER)
+
+    ApplicationEventScheduleFactory.create(
+        day=WeekdayChoice.MONDAY,
+        application_event=application.application_events.first(),
+    )
+
+    # when:
+    # - The user tries to allocate the schedule, but for a reservation unit
+    #   not in the event's event reservation units.
+    input_data = approve_data(application, begin="12:00:00", end="14:00:00")
+    response = graphql(APPROVE_MUTATION, input_data=input_data)
+
+    # then:
+    # - The response complains about the reservation unit not being included.
+    assert response.field_error_messages("allocatedDay") == [
+        "Cannot allocate multiple schedules on the same day for one event.",
+    ]
+
+
+def test_approve_schedule__cannot_approve_if_overlapping_with_another_approved_in_same_reservation_unit(graphql):
+    # given:
+    # - There is an allocatable application event schedule
+    # - There is an overlapping approved schedule for the same reservation unit
+    # - A superuser is using the system
+    reservation_unit = ReservationUnitFactory.create(spaces=[SpaceFactory.create()])
+    application = ApplicationFactory.create_application_ready_for_allocation(reservation_unit=reservation_unit)
+    ApplicationFactory.create_application_ready_for_allocation(
+        reservation_unit=reservation_unit,
+        pre_allocated=True,
+    )
+    graphql.login_user_based_on_type(UserType.SUPERUSER)
+
+    # when:
+    # - The user tries to allocate the schedule, but there is already
+    #   an allocation for the same reservation unit at that time.
+    input_data = approve_data(application, begin="10:00:00", end="11:00:00")
```

Comment:
Could test with the force flag to prove that it does not allow overriding this check.

Reply:
Done